File: src/test/java/fi/metatavu/metaform/server/test/functional/builder/impl/ApiTestBuilderResource.java
package fi.metatavu.metaform.server.test.functional.builder.impl;

import fi.metatavu.jaxrs.test.functional.builder.AbstractApiTestBuilderResource;
import fi.metatavu.jaxrs.test.functional.builder.AbstractTestBuilder;
import fi.metatavu.metaform.api.client.infrastructure.ApiClient;

/**
 * Abstract base class for API test resource builders
 *
 * @param <T> resource
 * @param <A> ApiClient for the resource
 */
abstract class ApiTestBuilderResource<T, A> extends AbstractApiTestBuilderResource<T, A, ApiClient> {

  private final ApiClient apiClient;

  /**
   * Returns API client
   *
   * @return API client
   */
  @Override
  protected ApiClient getApiClient() {
    return apiClient;
  }

  /**
   * Constructor
   *
   * @param testBuilder testBuilder
   */
  public ApiTestBuilderResource(AbstractTestBuilder<ApiClient> testBuilder, ApiClient apiClient) {
    super(testBuilder);
    this.apiClient = apiClient;
  }

}

File: src/test/java/fi/metatavu/metaform/server/test/functional/builder/impl/ApiTestBuilderResource.kt
package fi.metatavu.metaform.server.test.functional.builder.impl

import fi.metatavu.metaform.api.client.infrastructure.ApiClient
import fi.metatavu.metaform.api.client.infrastructure.ClientException
import fi.metatavu.metaform.api.client.infrastructure.ServerException
import fi.metatavu.metaform.server.test.functional.builder.TestBuilder
import org.junit.Assert

/**
 * Abstract base class for API test resource builders
 *
 * @author Jari Nykänen
 */
abstract class ApiTestBuilderResource<T, A>(
        protected val testBuilder: TestBuilder,
        private val apiClient: ApiClient
):fi.metatavu.jaxrs.test.functional.builder.AbstractAccessTokenApiTestBuilderResource<T, A, ApiClient>(testBuilder) {

    /**
     * Returns API client
     *
     * @return API client
     */
    override fun getApiClient(): ApiClient {
        return apiClient
    }

    /**
     * Asserts that client exception has expected status code
     *
     * @param expectedStatus expected status code
     * @param e client exception
     */
    protected fun assertClientExceptionStatus(expectedStatus: Int, e: ClientException) {
        Assert.assertEquals(expectedStatus, e.statusCode)
    }

    /**
     * Asserts that server exception has expected status code
     *
     * @param expectedStatus expected status code
     * @param e server exception
     */
    protected fun assertServerExceptionStatus(expectedStatus: Int, e: ServerException) {
        Assert.assertEquals(expectedStatus, e.statusCode)
    }
}